### DISCUSSION TITLE
Convert mask data

### DISCUSSION THREAD

**Discussion by peterhijma** [formkit/formkit#834] [General]
Thanks for this awesome form library. 🥳

When submitting the form, I'd like the input of a form field value to be converted. In my case I have this:

```js
        <FormKit
          type="mask"
          mask="##:##"
          name="foo"
        />
```

I want to convert the string to an integer. Ideally in a way I can also put some validation on that number.

Couldn't figure it out, tried to find anything in the docs.

**Comment by fenilli** [comment]
Validation can be done with custom validators, just convert to a number there and make the checks.

As for when submiting, you can make this conversion with the `submit` hook at the form node.

**Comment by peterhijma** [comment]
> Validation can be done with custom validators, just convert to a number there and make the checks.
> 
> As for when submiting, you can make this conversion with the `submit` hook at the form node.

@GustavoFenilli thanks for your response. I tried it with the `submit` hook. It did not work. :( 

I'm probably not doing the hooks right. I'll experiment a bit more and post the code later.

**Reply by fenilli** [reply]
For the `submit` hook, it is in the form itself, not at the field level.
```js
getNode('formid').hook.submit((payload, next) => {
  payload.foo = Number(payload.foo)
  next(payload)
})
```

**Comment by peterhijma** [comment]
> For the `submit` hook, it is in the form itself, not at the field level.
> 
> ```js
> getNode('formid').hook.submit((payload, next) => {
>   payload.foo = Number(payload.foo)
>   next(payload)
> })
> ```

That was it. Thanks a lot @GustavoFenilli

**Reply by andrew-boyd** [reply]
make sure you return the call to `next`

```
getNode('formid').hook.submit((payload, next) => {
  payload.foo = Number(payload.foo)
  return next(payload)
})
```
